components/date-picker/util.ts

```diff
@@ -160,3 +160,19 @@ export function getTimeProps<DateType, DisabledTime>(
     showTime: showTimeObj,
   };
 }
+
+type AllowClear = PickerProps<unknown>['allowClear'];
+type ClearIcon = PickerProps<unknown>['clearIcon'];
+
+export function mergeAllowClear(
+  allowClear: AllowClear,
+  clearIcon: ClearIcon,
+  defaultClearIcon: NonNullable<ClearIcon>,
+) {
+  if (allowClear === false) {
+    return false;
+  }
+
+  const defaults = { clearIcon: clearIcon ?? defaultClearIcon };
+  return typeof allowClear === 'object' && allowClear ? { ...defaults, ...allowClear } : defaults;
```

Comment:
```suggestion
  return typeof allowClear === 'object'  ? { ...defaults, ...allowClear } : defaults;
```
If `typeof allowClear === 'object'` holds, does that mean that allowClear must be truthy? Because our allowClear is non-null.

Reply:
That's right. Fixed.

I verified that this scenario is rejected by typescript:

```tsx
<DatePicker value={somepoint} allowClear={null} />
```